rxandroid/src/main/java/rx/android/app/AppObservable.java

```diff
@@ -93,17 +80,20 @@ public Boolean call(android.support.v4.app.Fragment fragment) {
      * @param fragment the fragment to bind the source sequence to
      * @param source   the source sequence
      */
-    public static <T> Observable<T> bindFragment(Object fragment, Observable<T> source) {
+    public static <T> Observable<T> bindFragment(Fragment fragment, Observable<T> source) {
         Assertions.assertUiThread();
-        final Observable<T> o = source.observeOn(mainThread());
-        if (USES_SUPPORT_FRAGMENTS && fragment instanceof android.support.v4.app.Fragment) {
-            android.support.v4.app.Fragment f = (android.support.v4.app.Fragment) fragment;
-            return o.lift(new OperatorConditionalBinding<T, android.support.v4.app.Fragment>(f, FRAGMENTV4_VALIDATOR));
-        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB && fragment instanceof Fragment) {
-            Fragment f = (Fragment) fragment;
-            return o.lift(new OperatorConditionalBinding<T, Fragment>(f, FRAGMENT_VALIDATOR));
-        } else {
-            throw new IllegalArgumentException("Target fragment is neither a native nor support library Fragment");
-        }
+        return source.observeOn(mainThread())
+                .lift(new OperatorConditionalBinding<T, Fragment>(fragment, FRAGMENT_VALIDATOR));
+    }
+
+    /**
+     * Binds the given source sequence to a support-v4 fragment.
+     *
+     * @see #bindFragment(Fragment, Observable)
+     */
+    public static <T> Observable<T> bindSupportFragment(android.support.v4.app.Fragment fragment, Observable<T> source) {
```

Comment:
Is there something that prevents using the same method name?
`public static <T> Observable<T> bindFragment(android.support.v4.app.Fragment fragment, Observable<T> source)`